### ISSUE TITLE
Curious to know remaining issues

### ISSUE THREAD

**Issue by dazinator** [bdukes/DNN-on-Docker#1]
Hi Brian!
Great work here. I am curious to know what you found in terms of remaining issues that would make this not yet ready for use as a local development approach? Sorry I wasnt able to catch your presentation at dnnsummit, I have never been able to attend as its too far (and too expensive) for me to get there!! :-)

**Comment by bdukes** [comment]
Thanks @dazinator, I appreciate your interest 😄 

- [ ] Cannot use a [bind mount](https://docs.docker.com/storage/bind-mounts/) with Docker for Windows which points to directory that exists in the container, e.g. `/bin` (this is a Windows containers issue)
- [ ] Inconsistent file change notification from IIS within a container, i.e. many times you will bind a local folder into a container, but it won't see your changes for minutes or longer (this appears to be an issue with IIS in Docker)
- [ ] DNN generates some URLs as absolute URLs, so port mapping doesn't work, e.g. you expose the site as `localhost:12345`, but the container sees it as `localhost:80` and doesn't generate URLs with the port, so links don't work (this is an issue within DNN)

**Comment by dazinator** [comment]
Thanks for the summary.

Am I right in thinking the first two issues are "dev workflow" related.
I am wondering.. if you deploy changes to the site via something like PollyDeploy (instead of `mounting`) perhaps FCN will work as then that alleviates the need to mount changes.

I know that you then lose the ability to edit files directly on the host that are shared with the container if doing it that way - is that what you were striving for?

**Comment by bdukes** [comment]
Yeah, that sounds right (I may have seen the FCN issues when editing files inside the container in general, but not sure).  If there's a means of continuously deploying changes to a site, that might become usable.

**Comment by dazinator** [comment]
> If there's a means of continuously deploying changes to a site, that might become usable.

The only way I can achieve that right now is at a module install level. E.g you make a change to a js file, and the whole module zip gets re-deployed.. Not sure if that will be good enough.

Otherwise mounting the /MyModule folder to /DesktopModules/MyModule would be a nice approach, but I assume this is what you had the problems with RE fcn?

**Comment by bdukes** [comment]
exactly 👍🏻

**Comment by BigDaddyKahuna** [comment]
While I've not tried mounting a web directory to an IIS container to facilitate updating dlls or .js files, I have, using Powershell, been able to copy files into/out of the web dir inside the container.     The Powershell method is cumbersome in that the container must be stopped for any file transfers to occur.

**Comment by dazinator** [comment]
@bdukes 
I found it interesting that you run the website on the host, and do the installation (install all the modules)  on the host first - before then adding the newly installed website and db to the container images. 

If you do install a vanilla website to the container with vanilla db, and go through the install wizard on the container - Dnn falls over running sql scripts. I managed to solve that issue by upping the container memory cap from 1gb to 2gb - the install wizard seems to utilise a fair amount of memory and by default the windows container can't cope with that! Did you come accross this?

**Comment by VashJuan** [comment]
There hopefully should be some future work on this topic at https://github.com/EOCOnline/OPLA/issues/1

**Comment by bogdan-litescu** [comment]
Hi @bdukes ! I was wondering if you have any news regarding DNN on Docker. 

I ran some experiments of my own with DNN 9.3.2. on Docker and got good behaviour.

I was able to mount a host folder using the -v parameter in docker run. I didn't experience any file sync issues so far. Did you have a specific scenario that was failing for you?

Also, because I connected directly to the container using its IP, I didn't run into the port mapping issues that you mentioned.

**Comment by bdukes** [comment]
@bogdan-litescu I haven't tried putting DNN in Docker again since I made note of these issues.  

Regarding file sync issues, I think the scenario that I was running into was trying to map a module's folder locally so that I can work on it.  IIS would stop seeing file changes that I made in the local folder.

**Comment by Artlake** [comment]
Hi guys, is it possible to containerize DNN website using Docker? I see that base image 'microsoft/aspnet' is not available anymore. Also i cant find any information that someone managed to run DNN website using docker containers. Thanks in advance.

**Comment by bdukes** [comment]
I'm not aware of anyone actively using Docker with DNN.
